json_tests/tests/test.rs

```diff
@@ -1133,6 +1133,8 @@ fn test_parse_enum_errors() {
             "trailing characters at line 1 column 9"),
         ("\"Frog\"",
             "EOF while parsing a value at line 1 column 6"),
+        ("\"Frog\" 0 ",
+            "EOF while parsing a value at line 1 column 6"),
```

Comment:
Hmm it is not clear to me why the `"\"Frog\""` case is an EOF error. It seems like that should be an invalid type error because the data contains a string and the enum expected a tuple variant. I will follow up on that.